docs/tools/terraform/reference/cookbook/postgres-clickhouse-integration-recipe.rst

```diff
@@ -0,0 +1,160 @@
+
+  2. ``variables.tf`` file
+
+  Use it for defining the variables to avoid including sensitive information in source control. The ``variables.tf`` file defines the API token, the project name, and the prefix for the service name.
+
+    .. code:: terraform
+
+       variable "aiven_api_token" {
+         description = "Aiven console API token"
+         type        = string
+       }
+   
+       variable "project_name" {
+         description = "Aiven console project name"
+         type        = string
+       }
+
+  3. ``*.tfvars`` file
+
+  Use it to indicate the actual values of the variables so that they can be passed (with the ``-var-file=`` flag) to Terraform during runtime and excluded later on. Configure the ``var-values.tfvars`` file as follows:
+
+    .. code-block:: terraform
+
+       aiven_api_token     = "<YOUR-AIVEN-AUTHENTICATION-TOKEN-GOES-HERE>"
+       project_name        = "<YOUR-AIVEN-CONSOLE-PROJECT-NAME-GOES-HERE>"
+
+Configure the ``services.tf`` file
+''''''''''''''''''''''''''''''''''
+
+The following Terraform script initializes both Aiven for PostgreSQL and Aiven for ClickHouse services, creates the service integration, the source PostgreSQL table, and the Aiven for ClickHouse database.
+
+.. code-block:: terraform
+
+  // Postgres service based in GCP US East
+  resource "aiven_pg" "postgres" {
+    project                 = var.project_name
+    service_name            = "postgres-gcp-us"
+    cloud_name              = "google-us-east-4"
+    plan                    = "business-8" // Primary + read replica
+    maintenance_window_dow  = "monday"
+    maintenance_window_time = "10:00:00"
+  }
+
+  // Postgres sensors database
+  resource "aiven_pg_database" "mydatabase" {
+    project       = var.project_name
+    service_name  = aiven_pg.postgres.service_name
+    database_name = "sensors"
+  }
+
+  // ClickHouse service based in the same region
+  resource "aiven_clickhouse" "clickhouse" {
+    project                 = var.project_name
+    service_name            = "clickhouse-gcp-us"
+    cloud_name              = "google-us-east-4"
+    plan                    = "startup-4"
+    maintenance_window_dow  = "monday"
+    maintenance_window_time = "10:00:00"
+  }
+
+  // Sample ClickHouse database that can be used to write and process raw data
+  resource "aiven_clickhouse_database" "iot_measurements" {
+    project      = var.project_name
+    service_name = aiven_clickhouse.clickhouse.service_name
+    name         = "iot_measurements"
+  }
+
+  // ClickHouse service integration for the PostgreSQL service as a source
+  resource "aiven_service_integration" "clickhouse_postgres_source" {
+    project                  = var.project_name
+    integration_type         = "clickhouse_postgresql"
+    source_service_name      = aiven_pg.postgres.service_name
+    destination_service_name = aiven_clickhouse.clickhouse.service_name
```

Comment:
Here we need to add a configuration block that has been surfaced by the latest terraform provider, for instance

```suggestion
    destination_service_name = aiven_clickhouse.clickhouse.service_name
    clickhouse_postgresql_user_config {
      databases {
        database = aiven_pg_database. sensor_dims.database_name
        schema = "public"
      }
  }
```